AudioPlayer/AudioPlayer/AudioPlayer.swift

```diff
@@ -601,11 +601,70 @@ public class AudioPlayer: NSObject {
     - parameter time: The time to seek to.
     */
     public func seekToTime(time: NSTimeInterval) {
-        player?.seekToTime(CMTimeMake(Int64(time), 1))
+        let time = CMTime(seconds: time, preferredTimescale: 1000000000)
+        let seekableRange = player?.currentItem?.seekableTimeRanges.last?.CMTimeRangeValue
+        if let seekableStart = seekableRange?.start, let seekableEnd = seekableRange?.end {
+            // check if time is in seekable range
+            if time >= seekableStart && time <= seekableEnd {
+                // time is in seekable range
+                player?.seekToTime(time)
+            }
+            else if time < seekableStart {
+                // time is before seekable start, so just move to the most early position as possible
+                seekToSeekableRangeStart()
+            }
+            else if time > seekableEnd {
+                // time is larger than possibly, so just move forward as far as possible
+                seekToSeekableRangeEnd()
+            }
+            
+            updateNowPlayingInfoCenter()
+        }
+    }
+    
+    /**
+     Seeks forward as far as possible
+     
+     */
+    public func seekToSeekableRangeEnd() {
+        let bufferTime = CMTime(seconds: 1, preferredTimescale: 1000000000)
+        let (earliesPoint, latestPoint) = getSeekableBordersWithBufferTime(bufferTime)
+        let newPos = max(earliesPoint, latestPoint)
+        
+        player?.seekToTime(newPos)
         updateNowPlayingInfoCenter()
     }
 
     /**
+     Seeks backwards as far as possible.
+     
+     */
+    public func seekToSeekableRangeStart() {
+        let bufferTime = CMTime(seconds: 1, preferredTimescale: 1000000000)
+        let (earliesPoint, latestPoint) = getSeekableBordersWithBufferTime(bufferTime)
+        let newPos = min(earliesPoint, latestPoint)
+        
+        player?.seekToTime(newPos)
+        updateNowPlayingInfoCenter()
+    }
+    
+    /**
+     Returns the earliest and latest possible point of the current seekable range with a margin to
+     the actual borders of 1 second.
+
+    - parameter bufferTime: set the margin buffer time of the latest point to the end of the actual
+                            seekable range. A minimum of 1 second will be enforced.
+    */
+    private func getSeekableBordersWithBufferTime(var bufferTime: CMTime) -> (earliesPoint: CMTime, latestPoint: CMTime) {
+        let marginBuffer = CMTime(seconds: 1, preferredTimescale: 1000000000)
+        bufferTime = max(bufferTime, marginBuffer)
+        let seekableRange = player?.currentItem?.seekableTimeRanges.last?.CMTimeRangeValue
+        let latestPoint = max(seekableRange!.start, seekableRange!.end - bufferTime)
+        let earliesPoint = min(seekableRange!.end, seekableRange!.start + marginBuffer)
```

Comment:
Can you do the same here for forced unwrap? User can call seekToSeekableXXX and get an unexpected crash there.